modules/invisiblyAnalyticsAdapter.js

```diff
@@ -55,9 +55,26 @@ let _pageView = {
   language: window.navigator.language,
   vendor: window.navigator.vendor,
   screenWidth: x,
-  screenHeight: y
+  screenHeight: y,
 };
 
+let weightedFilter = (function () {
```

Comment:
With just `Math.random() > 0.99`, the accuracy is not that high, so though of looping through the pass/fail specs. 
Let us know your thoughts on this.

Reply:
I did some experiments with `Math.random()`:

```
var counter = 0;
for (var i = 0; i < 100000000; i++) {
    counter += Number(Math.random() > 0.99);
}

console.log(counter);`
```

And here are my results:

```
1000560
999359
1000474
1000116
1000180
999773
```

So, I would say that for you case distribution is 'pretty good'.  

And for your looping through the pass/fail specs, in your first iteration of `for` loop you will compare `0.99 <= Math.random()` (because `fail` key will be first in loop). So, bottom line, you are still using `Math.random()` to filter out 99% cases.

Reply:
Have updated the filter logic as per your recommendation.